packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart

```diff
@@ -175,11 +175,33 @@ class MethodChannelCamera extends CameraPlatform {
         <String, dynamic>{'cameraId': cameraId},
       );
 
+  @override
+  Future<void> setFlashMode(int cameraId, FlashMode mode) =>
+      _channel.invokeMethod<void>(
+        'setFlashMode',
+        <String, dynamic>{
+          'cameraId': cameraId,
+          'mode': _serializeFlashMode(mode),
+        },
+      );
+
   @override
   Widget buildPreview(int cameraId) {
     return Texture(textureId: cameraId);
   }
 
+  /// Returns the flash mode as a String.
+  String _serializeFlashMode(FlashMode flashMode) {
+    switch (flashMode) {
+      case FlashMode.off:
+        return 'off';
+      case FlashMode.auto:
+        return 'auto';
+      case FlashMode.always:
+        return 'always';
+    }
+  }
```

Comment:
You should handle the `default` case and throw an argument error explaining the supplied value cannot be supported.